tools/src/main/java/org/apache/kafka/trogdor/coordinator/CoordinatorClient.java

```diff
@@ -170,111 +193,291 @@ public void shutdown() throws Exception {
+                TaskRequest req = new TaskRequest(taskId);
+                TaskState taskState = null;
+                try {
+                    taskState = client.task(req);
+                } catch (NotFoundException e) {
+                    System.out.printf("Task %s was not found.%n", taskId);
+                    Exit.exit(1);
+                }
+                if (res.getBoolean("json")) {
+                    System.out.println(JsonUtil.toJsonString(taskState));
+                } else {
+                    System.out.printf("Task %s of type %s is %s. %s%n", taskId,
+                        taskState.spec().getClass().getCanonicalName(),
+                        taskState.stateType(), prettyPrintTaskInfo(taskState, localOffset));
+                    if (taskState instanceof TaskDone) {
+                        TaskDone taskDone = (TaskDone) taskState;
+                        if ((taskDone.error() != null) && (!taskDone.error().isEmpty())) {
+                            System.out.printf("Error: %s%n", taskDone.error());
+                        }
+                    }
+                    if (res.getBoolean("verbose")) {
+                        System.out.printf("Spec: %s%n%n", JsonUtil.toPrettyJsonString(taskState.spec()));
+                    }
+                    if (res.getBoolean("verbose") || res.getBoolean("showStatus")) {
+                        System.out.printf("Status: %s%n%n", JsonUtil.toPrettyJsonString(taskState.status()));
+                    }
+                }
+                break;
+            }
+            case "showTasks": {
+                TaskStateType taskStateType = res.<TaskStateType>get("taskStateType");
+                List<String> taskIds = new ArrayList<>();
+                Pattern taskIdPattern = null;
+                if (res.getList("taskIds") != null) {
+                    for (Object taskId : res.getList("taskIds")) {
+                        taskIds.add((String) taskId);
+                    }
+                } else if (res.getString("taskIdPattern") != null) {
+                    try {
+                        taskIdPattern = Pattern.compile(res.getString("taskIdPattern"));
+                    } catch (PatternSyntaxException e) {
+                        System.out.println("Invalid task ID regular expression " + res.getString("taskIdPattern"));
+                        e.printStackTrace();
+                        Exit.exit(1);
+                    }
+                }
+                TasksRequest req = new TasksRequest(taskIds, 0, 0, 0, 0,
+                    Optional.ofNullable(taskStateType));
+                TasksResponse response = client.tasks(req);
+                if (taskIdPattern != null) {
+                    TreeMap<String, TaskState> filteredTasks = new TreeMap<>();
+                    for (Map.Entry<String, TaskState> entry : response.tasks().entrySet()) {
+                        if (taskIdPattern.matcher(entry.getKey()).matches()) {
```

Comment:
This is great functionality but would it be better if we had it in the coordinator server in order to leverage it from REST API clients in other languages? Otherwise we probably need to duplicate logic.
 We could maybe open a JIRA to track this as a separate item